src/YieldDisburser.sol

```diff
@@ -467,4 +414,14 @@ contract YieldDisburser is OwnableUpgradeable {
 
         blockTime = _blockTime;
```

Comment:
`setBlockTime` serves no purpose in this contract. The variable is only used in the "initializer" as a constant to calculate initial values. This function should definitely be removed and "blockTime" should be removed as a storage variable entirely.

I'd recommend trying to remove it from the calculations entirely and change the approach to explicitly defining the initial values of variables instead of calculating them in the "initailizer". There's not really a reason to not just pre-calculate the "cycleLength" for example. Same goes for "minRequiredVotingPower" where all the other values in the equation are know before the contract is initialized.

Reply:
addressed by 9e6a03a